cd .../fastq
umi_tools extract --extract-method=regex --bc-pattern2="^(?P<discard_1>GGTGTAGTGGGTTTGGAGG){s<=3}.{11}(?P<discard_2>.{30}).{11}(?P<discard_3>.{30})(?P<umi_1>.)(?P<discard_4>.*)$" -I /mnt/server1/data2/tangym/p300s/tangym/Analysis_data/dir/sample_name_R1.fq.gz -S .../fastq/sample_name.extract_R1.fq.gz --read2-in=/mnt/server1/data2/tangym/p300s/tangym/Analysis_data/dir/sample_name_R2.fq.gz --read2-out=.../fastq/sample_name.extract_R2.fq.gz -L extract.log && echo -e "umi_tools2\tWed May  1 09:32:09 2024\tumi_tools extract --extract-method=regex --bc-pattern2=\x22^(?P<discard_1>GGTGTAGTGGGTTTGGAGG){s<=3}.{11}(?P<discard_2>.{30}).{11}(?P<discard_3>.{30})(?P<umi_1>.)(?P<discard_4>.*)$\x22 -I /mnt/server1/data2/tangym/p300s/tangym/Analysis_data/dir/sample_name_R1.fq.gz -S .../fastq/sample_name.extract_R1.fq.gz --read2-in=/mnt/server1/data2/tangym/p300s/tangym/Analysis_data/dir/sample_name_R2.fq.gz --read2-out=.../fastq/sample_name.extract_R2.fq.gz -L extract.log #ok\n\n\n" >> readme.status.txt
cd .../fastq
zcat .../fastq/sample_name.extract_R2.fq.gz | awk 'NR%4==2{print substr($1,1,22)}' > sample_name.extract_R2.fq.barcode && echo -e "umi_tools_zcat.sample_name.extract_R2.fq.gz\tWed May  1 09:32:09 2024\tzcat .../fastq/sample_name.extract_R2.fq.gz | awk \x27NR%4==2{print substr($1,1,22)}\x27 > sample_name.extract_R2.fq.barcode #ok\n\n\n" >> readme.status.txt
cd .../fastq
less sample_name.extract_R2.fq.barcode | awk -F "\t" '{$2=NR; print $0}' OFS="\t" > sample_name.extract_R2.fq.barcode.order && echo -e "or2_order.sample_name.extract_R2.fq.gz\tWed May  1 09:32:09 2024\tless sample_name.extract_R2.fq.barcode | awk -F \x22\t\x22 \x27{$2=NR; print $0}\x27 OFS=\x22\t\x22 > sample_name.extract_R2.fq.barcode.order #ok\n\n\n" >> readme.status.txt
cd .../fastq
/home/liulab/anaconda3/envs/tangym_workspace/bin/Rscript /mnt/server1/data2/tangym/p300s/tangym/Analysis_data/auto_spatial_methylation/bins/2_check_barcode.R -b /mnt/server1/data2/tangym/p300s/tangym/Analysis_data/auto_spatial_methylation/bins/combine_barcode.round2round1_index1_index2.v3_big_methylation.txt -p . -n sample_name.extract_R2.fq.barcode && echo -e "2_check_barcode.R.sample_name.extract_R2.fq.gz\tWed May  1 09:32:09 2024\t/home/liulab/anaconda3/envs/tangym_workspace/bin/Rscript /mnt/server1/data2/tangym/p300s/tangym/Analysis_data/auto_spatial_methylation/bins/2_check_barcode.R -b /mnt/server1/data2/tangym/p300s/tangym/Analysis_data/auto_spatial_methylation/bins/combine_barcode.round2round1_index1_index2.v3_big_methylation.txt -p . -n sample_name.extract_R2.fq.barcode #ok\n\n\n" >> readme.status.txt
cd .../fastq
zcat .../fastq/sample_name.extract_R2.fq.gz | awk 'NR%4==1{print $0}' | cut -f1 -d " " | cut -f1 -d "_" > sample_name.extract_R2.fq.gz.filter.1_col.txt && echo -e "zcatawkcut.01_get_read2.sh..sample_name.extract_R2.fq.gz\tWed May  1 09:32:09 2024\tzcat .../fastq/sample_name.extract_R2.fq.gz | awk \x27NR%4==1{print $0}\x27 | cut -f1 -d \x22 \x22 | cut -f1 -d \x22_\x22 > sample_name.extract_R2.fq.gz.filter.1_col.txt #ok\n\n\n" >> readme.status.txt
cd .../fastq
zcat .../fastq/sample_name.raw_R2.fq.gz | awk 'NR%4==1{print $0}' | cut -f1 -d " " | cut -f1 -d "_" > sample_name.raw_R2.fq.gz.filter.1_col.txt && echo -e "zcatawkcut.01_get_read2.sh..sample_name.raw_R2.fq.gz\tWed May  1 09:32:09 2024\tzcat .../fastq/sample_name.raw_R2.fq.gz | awk \x27NR%4==1{print $0}\x27 | cut -f1 -d \x22 \x22 | cut -f1 -d \x22_\x22 > sample_name.raw_R2.fq.gz.filter.1_col.txt #ok\n\n\n" >> readme.status.txt
cd .../fastq
less sample_name.raw_R2.fq.gz.filter.1_col.txt | awk 'NR==FNR{a[$1]=NR}NR!=FNR{print a[$1]}' - sample_name.extract_R2.fq.gz.filter.1_col.txt | grep -v '^$' > sample_name.extract_R2.fq.gz.1_col.filter.order.txt && echo -e "filter_1_col.order.01_get_read2.sh\tWed May  1 09:32:09 2024\tless sample_name.raw_R2.fq.gz.filter.1_col.txt | awk \x27NR==FNR{a[$1]=NR}NR!=FNR{print a[$1]}\x27 - sample_name.extract_R2.fq.gz.filter.1_col.txt | grep -v \x27^$\x27 > sample_name.extract_R2.fq.gz.1_col.filter.order.txt #ok\n\n\n" >> readme.status.txt
cd .../fastq
less sample_name.extract_R2.fq.gz.1_col.filter.order.txt | awk -F "\t" '{$2=($1-1)*4+ 1 ; print $2}' > sample_name.extract_R2.fq.gz.1_col.filter.order.txt.order_1 && echo -e "filter_1_col.order.01_get_read2.sh.sample_name.extract_R2.fq.gz.1_col.filter.order.txt.order_1\tWed May  1 09:32:09 2024\tless sample_name.extract_R2.fq.gz.1_col.filter.order.txt | awk -F \x22\t\x22 \x27{$2=($1-1)*4+ 1 ; print $2}\x27 > sample_name.extract_R2.fq.gz.1_col.filter.order.txt.order_1 #ok\n\n\n" >> readme.status.txt
cd .../fastq
python3 /mnt/server1/data2/tangym/p300s/tangym/Analysis_data/auto_spatial_methylation/python_script/fq_line_pick_and_gz.py sample_name.extract_R2.fq.gz.1_col.filter.order.txt.order_1 .../fastq/sample_name.raw_R2.fq.gz -o raw_R2.sample_name.extract_R2.extract_all.fq.gz && echo -e "extract_order_all_fq_gz.01_get_read2.sh\tWed May  1 09:32:09 2024\tpython3 /mnt/server1/data2/tangym/p300s/tangym/Analysis_data/auto_spatial_methylation/python_script/fq_line_pick_and_gz.py sample_name.extract_R2.fq.gz.1_col.filter.order.txt.order_1 .../fastq/sample_name.raw_R2.fq.gz -o raw_R2.sample_name.extract_R2.extract_all.fq.gz #ok\n\n\n" >> readme.status.txt
#!!!Now is: raw_R2.sample_name.extract_R2.extract_all.fq.gz watson
cd .../fastq
umi_tools extract --extract-method=regex --bc-pattern="^(?P<discard_1>GGTGTAGTGGGTTTGGAGG){s<=3}(?P<cell_1>.{11})(?P<discard_2>..CC.C...C.....C.C.C..C...C...){s<=4}(?P<cell_2>.{11})(?P<discard_3>CCC.....C..CC....C...C...CC...){s<=4}(?P<discard_4>AGATGTGT){s<=2}(?P<umi_1>.)(?P<discard_5>.*)$" -I raw_R2.sample_name.extract_R2.extract_all.fq.gz -S raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz -L raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz.log && echo -e "raw_R2.sample_name.extract_R2.extract_all.fq.gzwatson\tWed May  1 09:32:09 2024\tumi_tools extract --extract-method=regex --bc-pattern=\x22^(?P<discard_1>GGTGTAGTGGGTTTGGAGG){s<=3}(?P<cell_1>.{11})(?P<discard_2>..CC.C...C.....C.C.C..C...C...){s<=4}(?P<cell_2>.{11})(?P<discard_3>CCC.....C..CC....C...C...CC...){s<=4}(?P<discard_4>AGATGTGT){s<=2}(?P<umi_1>.)(?P<discard_5>.*)$\x22 -I raw_R2.sample_name.extract_R2.extract_all.fq.gz -S raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz -L raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz.log #ok\n\n\n" >> readme.status.txt
cd .../fastq
zcat raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz | awk 'NR%4==1{print $0}' | cut -f1 -d " " | cut -f1 -d "_" > raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz.filter.1_col.txt && echo -e "zcatawkcut.03_get_read2_watson.sh.raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz\tWed May  1 09:32:09 2024\tzcat raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz | awk \x27NR%4==1{print $0}\x27 | cut -f1 -d \x22 \x22 | cut -f1 -d \x22_\x22 > raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz.filter.1_col.txt #ok\n\n\n" >> readme.status.txt
cd .../fastq
zcat .../fastq/sample_name.extract_R2.fq.gz | awk 'NR%4==1{print $0}' | cut -f1 -d " " | cut -f1 -d "_" > sample_name.extract_R2.fq.gz.filter.1_col.txt && echo -e "zcatawkcut.03_get_read2_watson.sh..sample_name.extract_R2.fq.gz\tWed May  1 09:32:09 2024\tzcat .../fastq/sample_name.extract_R2.fq.gz | awk \x27NR%4==1{print $0}\x27 | cut -f1 -d \x22 \x22 | cut -f1 -d \x22_\x22 > sample_name.extract_R2.fq.gz.filter.1_col.txt #ok\n\n\n" >> readme.status.txt
cd .../fastq
less sample_name.extract_R2.fq.gz.filter.1_col.txt | awk 'NR==FNR{a[$1]=NR}NR!=FNR{print a[$1]}' - raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz.filter.1_col.txt | grep -v '^$' > raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz.1_col.filter.order.txt && echo -e "filter_1_col.order.03_get_read2_watson.sh\tWed May  1 09:32:09 2024\tless sample_name.extract_R2.fq.gz.filter.1_col.txt | awk \x27NR==FNR{a[$1]=NR}NR!=FNR{print a[$1]}\x27 - raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz.filter.1_col.txt | grep -v \x27^$\x27 > raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz.1_col.filter.order.txt #ok\n\n\n" >> readme.status.txt
cd .../fastq
less raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz.1_col.filter.order.txt | awk -F "\t" '{$2=($1-1)*4+ 1 ; print $2}' > raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz.1_col.filter.order.txt.order_1 && echo -e "filter_1_col.order.03_get_read2_watson.sh.raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz.1_col.filter.order.txt.order_1\tWed May  1 09:32:09 2024\tless raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz.1_col.filter.order.txt | awk -F \x22\t\x22 \x27{$2=($1-1)*4+ 1 ; print $2}\x27 > raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz.1_col.filter.order.txt.order_1 #ok\n\n\n" >> readme.status.txt
cd .../fastq
python3 /mnt/server1/data2/tangym/p300s/tangym/Analysis_data/auto_spatial_methylation/python_script/fq_line_pick_and_gz.py raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz.1_col.filter.order.txt.order_1 .../fastq/sample_name.extract_R2.fq.gz -o raw_R2.sample_name.extract_R2.extract_watson.fq.gz && echo -e "extract_order_all_fq_gz.03_get_read2_watson.sh\tWed May  1 09:32:09 2024\tpython3 /mnt/server1/data2/tangym/p300s/tangym/Analysis_data/auto_spatial_methylation/python_script/fq_line_pick_and_gz.py raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz.1_col.filter.order.txt.order_1 .../fastq/sample_name.extract_R2.fq.gz -o raw_R2.sample_name.extract_R2.extract_watson.fq.gz #ok\n\n\n" >> readme.status.txt
cd .../fastq
zcat raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz | awk 'NR%4==1{print $0}' | cut -f1 -d " " | cut -f1 -d "_" > raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz.filter.1_col.txt && echo -e "zcatawkcut.04_get_watson.sh.raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz\tWed May  1 09:32:09 2024\tzcat raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz | awk \x27NR%4==1{print $0}\x27 | cut -f1 -d \x22 \x22 | cut -f1 -d \x22_\x22 > raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz.filter.1_col.txt #ok\n\n\n" >> readme.status.txt
cd .../fastq
zcat .../fastq/sample_name.raw_R2.fq.gz | awk 'NR%4==1{print $0}' | cut -f1 -d " " | cut -f1 -d "_" > sample_name.raw_R2.fq.gz.filter.1_col.txt && echo -e "zcatawkcut.04_get_watson.sh..sample_name.raw_R2.fq.gz\tWed May  1 09:32:09 2024\tzcat .../fastq/sample_name.raw_R2.fq.gz | awk \x27NR%4==1{print $0}\x27 | cut -f1 -d \x22 \x22 | cut -f1 -d \x22_\x22 > sample_name.raw_R2.fq.gz.filter.1_col.txt #ok\n\n\n" >> readme.status.txt
cd .../fastq
less sample_name.raw_R2.fq.gz.filter.1_col.txt | awk 'NR==FNR{a[$1]=NR}NR!=FNR{print a[$1]}' - raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz.filter.1_col.txt | grep -v '^$' > raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz.1_col.filter.order.txt && echo -e "filter_1_col.order.04_get_watson.sh\tWed May  1 09:32:09 2024\tless sample_name.raw_R2.fq.gz.filter.1_col.txt | awk \x27NR==FNR{a[$1]=NR}NR!=FNR{print a[$1]}\x27 - raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz.filter.1_col.txt | grep -v \x27^$\x27 > raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz.1_col.filter.order.txt #ok\n\n\n" >> readme.status.txt
cd .../fastq
less raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz.1_col.filter.order.txt | awk -F "\t" '{$2=($1-1)*4+ 1 ; print $2}' > raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz.1_col.filter.order.txt.order_1 && echo -e "filter_1_col.order.04_get_watson.sh.raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz.1_col.filter.order.txt.order_1\tWed May  1 09:32:09 2024\tless raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz.1_col.filter.order.txt | awk -F \x22\t\x22 \x27{$2=($1-1)*4+ 1 ; print $2}\x27 > raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz.1_col.filter.order.txt.order_1 #ok\n\n\n" >> readme.status.txt
cd .../fastq
python3 /mnt/server1/data2/tangym/p300s/tangym/Analysis_data/auto_spatial_methylation/python_script/fq_line_pick_and_gz.py raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz.1_col.filter.order.txt.order_1 .../fastq/sample_name.raw_R2.fq.gz -o raw_R2.sample_name.extract_R2.extract_watson.fq.gz && echo -e "extract_order_all_fq_gz.04_get_watson.sh\tWed May  1 09:32:09 2024\tpython3 /mnt/server1/data2/tangym/p300s/tangym/Analysis_data/auto_spatial_methylation/python_script/fq_line_pick_and_gz.py raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz.1_col.filter.order.txt.order_1 .../fastq/sample_name.raw_R2.fq.gz -o raw_R2.sample_name.extract_R2.extract_watson.fq.gz #ok\n\n\n" >> readme.status.txt
cd .../fastq
zcat raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz | awk 'NR%4==1{print $0}' | cut -f1 -d " " | cut -f1 -d "_" > raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz.filter.1_col.txt && echo -e "zcatawkcut.04_get_watson.sh.raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz\tWed May  1 09:32:09 2024\tzcat raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz | awk \x27NR%4==1{print $0}\x27 | cut -f1 -d \x22 \x22 | cut -f1 -d \x22_\x22 > raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz.filter.1_col.txt #ok\n\n\n" >> readme.status.txt
cd .../fastq
zcat .../fastq/sample_name.raw_R1.fq.gz | awk 'NR%4==1{print $0}' | cut -f1 -d " " | cut -f1 -d "_" > sample_name.raw_R1.fq.gz.filter.1_col.txt && echo -e "zcatawkcut.04_get_watson.sh..sample_name.raw_R1.fq.gz\tWed May  1 09:32:09 2024\tzcat .../fastq/sample_name.raw_R1.fq.gz | awk \x27NR%4==1{print $0}\x27 | cut -f1 -d \x22 \x22 | cut -f1 -d \x22_\x22 > sample_name.raw_R1.fq.gz.filter.1_col.txt #ok\n\n\n" >> readme.status.txt
cd .../fastq
less sample_name.raw_R1.fq.gz.filter.1_col.txt | awk 'NR==FNR{a[$1]=NR}NR!=FNR{print a[$1]}' - raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz.filter.1_col.txt | grep -v '^$' > raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz.1_col.filter.order.txt && echo -e "filter_1_col.order.04_get_watson.sh\tWed May  1 09:32:09 2024\tless sample_name.raw_R1.fq.gz.filter.1_col.txt | awk \x27NR==FNR{a[$1]=NR}NR!=FNR{print a[$1]}\x27 - raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz.filter.1_col.txt | grep -v \x27^$\x27 > raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz.1_col.filter.order.txt #ok\n\n\n" >> readme.status.txt
cd .../fastq
less raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz.1_col.filter.order.txt | awk -F "\t" '{$2=($1-1)*4+ 1 ; print $2}' > raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz.1_col.filter.order.txt.order_1 && echo -e "filter_1_col.order.04_get_watson.sh.raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz.1_col.filter.order.txt.order_1\tWed May  1 09:32:09 2024\tless raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz.1_col.filter.order.txt | awk -F \x22\t\x22 \x27{$2=($1-1)*4+ 1 ; print $2}\x27 > raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz.1_col.filter.order.txt.order_1 #ok\n\n\n" >> readme.status.txt
cd .../fastq
python3 /mnt/server1/data2/tangym/p300s/tangym/Analysis_data/auto_spatial_methylation/python_script/fq_line_pick_and_gz.py raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz.1_col.filter.order.txt.order_1 .../fastq/sample_name.raw_R1.fq.gz -o raw_R1.raw_R2.sample_name.extract_R2.extract_watson.fq.gz && echo -e "extract_order_all_fq_gz.04_get_watson.sh\tWed May  1 09:32:09 2024\tpython3 /mnt/server1/data2/tangym/p300s/tangym/Analysis_data/auto_spatial_methylation/python_script/fq_line_pick_and_gz.py raw_R2.sample_name.extract_R2.extract_all.watson.fq.gz.1_col.filter.order.txt.order_1 .../fastq/sample_name.raw_R1.fq.gz -o raw_R1.raw_R2.sample_name.extract_R2.extract_watson.fq.gz #ok\n\n\n" >> readme.status.txt
#!!!Now is: raw_R2.sample_name.extract_R2.extract_all.fq.gz watson End!!

#!!!Now is: raw_R2.sample_name.extract_R2.extract_all.fq.gz crick
cd .../fastq
umi_tools extract --extract-method=regex --bc-pattern="^(?P<discard_1>GGTGTAGTGGGTTTGGAGG){s<=3}(?P<cell_1>.{11})(?P<discard_2>..TT.T...T.....T.T.T..T...T...){s<=4}(?P<cell_2>.{11})(?P<discard_3>TTT.....T..TT....T...T...TT...){s<=4}(?P<discard_4>AGATGTGT){s<=2}(?P<umi_1>.)(?P<discard_5>.*)$" -I raw_R2.sample_name.extract_R2.extract_all.fq.gz -S raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz -L raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz.log && echo -e "raw_R2.sample_name.extract_R2.extract_all.fq.gzcrick\tWed May  1 09:32:09 2024\tumi_tools extract --extract-method=regex --bc-pattern=\x22^(?P<discard_1>GGTGTAGTGGGTTTGGAGG){s<=3}(?P<cell_1>.{11})(?P<discard_2>..TT.T...T.....T.T.T..T...T...){s<=4}(?P<cell_2>.{11})(?P<discard_3>TTT.....T..TT....T...T...TT...){s<=4}(?P<discard_4>AGATGTGT){s<=2}(?P<umi_1>.)(?P<discard_5>.*)$\x22 -I raw_R2.sample_name.extract_R2.extract_all.fq.gz -S raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz -L raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz.log #ok\n\n\n" >> readme.status.txt
cd .../fastq
zcat raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz | awk 'NR%4==1{print $0}' | cut -f1 -d " " | cut -f1 -d "_" > raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz.filter.1_col.txt && echo -e "zcatawkcut.03_get_read2_crick.sh.raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz\tWed May  1 09:32:09 2024\tzcat raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz | awk \x27NR%4==1{print $0}\x27 | cut -f1 -d \x22 \x22 | cut -f1 -d \x22_\x22 > raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz.filter.1_col.txt #ok\n\n\n" >> readme.status.txt
cd .../fastq
zcat .../fastq/sample_name.extract_R2.fq.gz | awk 'NR%4==1{print $0}' | cut -f1 -d " " | cut -f1 -d "_" > sample_name.extract_R2.fq.gz.filter.1_col.txt && echo -e "zcatawkcut.03_get_read2_crick.sh..sample_name.extract_R2.fq.gz\tWed May  1 09:32:09 2024\tzcat .../fastq/sample_name.extract_R2.fq.gz | awk \x27NR%4==1{print $0}\x27 | cut -f1 -d \x22 \x22 | cut -f1 -d \x22_\x22 > sample_name.extract_R2.fq.gz.filter.1_col.txt #ok\n\n\n" >> readme.status.txt
cd .../fastq
less sample_name.extract_R2.fq.gz.filter.1_col.txt | awk 'NR==FNR{a[$1]=NR}NR!=FNR{print a[$1]}' - raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz.filter.1_col.txt | grep -v '^$' > raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz.1_col.filter.order.txt && echo -e "filter_1_col.order.03_get_read2_crick.sh\tWed May  1 09:32:09 2024\tless sample_name.extract_R2.fq.gz.filter.1_col.txt | awk \x27NR==FNR{a[$1]=NR}NR!=FNR{print a[$1]}\x27 - raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz.filter.1_col.txt | grep -v \x27^$\x27 > raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz.1_col.filter.order.txt #ok\n\n\n" >> readme.status.txt
cd .../fastq
less raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz.1_col.filter.order.txt | awk -F "\t" '{$2=($1-1)*4+ 1 ; print $2}' > raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz.1_col.filter.order.txt.order_1 && echo -e "filter_1_col.order.03_get_read2_crick.sh.raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz.1_col.filter.order.txt.order_1\tWed May  1 09:32:09 2024\tless raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz.1_col.filter.order.txt | awk -F \x22\t\x22 \x27{$2=($1-1)*4+ 1 ; print $2}\x27 > raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz.1_col.filter.order.txt.order_1 #ok\n\n\n" >> readme.status.txt
cd .../fastq
python3 /mnt/server1/data2/tangym/p300s/tangym/Analysis_data/auto_spatial_methylation/python_script/fq_line_pick_and_gz.py raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz.1_col.filter.order.txt.order_1 .../fastq/sample_name.extract_R2.fq.gz -o raw_R2.sample_name.extract_R2.crick.extract_all.fq.gz && echo -e "extract_order_all_fq_gz.03_get_read2_crick.sh\tWed May  1 09:32:09 2024\tpython3 /mnt/server1/data2/tangym/p300s/tangym/Analysis_data/auto_spatial_methylation/python_script/fq_line_pick_and_gz.py raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz.1_col.filter.order.txt.order_1 .../fastq/sample_name.extract_R2.fq.gz -o raw_R2.sample_name.extract_R2.crick.extract_all.fq.gz #ok\n\n\n" >> readme.status.txt
cd .../fastq
zcat raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz | awk 'NR%4==1{print $0}' | cut -f1 -d " " | cut -f1 -d "_" > raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz.filter.1_col.txt && echo -e "zcatawkcut.04_get_crick.sh.raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz\tWed May  1 09:32:09 2024\tzcat raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz | awk \x27NR%4==1{print $0}\x27 | cut -f1 -d \x22 \x22 | cut -f1 -d \x22_\x22 > raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz.filter.1_col.txt #ok\n\n\n" >> readme.status.txt
cd .../fastq
zcat .../fastq/sample_name.raw_R2.fq.gz | awk 'NR%4==1{print $0}' | cut -f1 -d " " | cut -f1 -d "_" > sample_name.raw_R2.fq.gz.filter.1_col.txt && echo -e "zcatawkcut.04_get_crick.sh..sample_name.raw_R2.fq.gz\tWed May  1 09:32:09 2024\tzcat .../fastq/sample_name.raw_R2.fq.gz | awk \x27NR%4==1{print $0}\x27 | cut -f1 -d \x22 \x22 | cut -f1 -d \x22_\x22 > sample_name.raw_R2.fq.gz.filter.1_col.txt #ok\n\n\n" >> readme.status.txt
cd .../fastq
less sample_name.raw_R2.fq.gz.filter.1_col.txt | awk 'NR==FNR{a[$1]=NR}NR!=FNR{print a[$1]}' - raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz.filter.1_col.txt | grep -v '^$' > raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz.1_col.filter.order.txt && echo -e "filter_1_col.order.04_get_crick.sh\tWed May  1 09:32:09 2024\tless sample_name.raw_R2.fq.gz.filter.1_col.txt | awk \x27NR==FNR{a[$1]=NR}NR!=FNR{print a[$1]}\x27 - raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz.filter.1_col.txt | grep -v \x27^$\x27 > raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz.1_col.filter.order.txt #ok\n\n\n" >> readme.status.txt
cd .../fastq
less raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz.1_col.filter.order.txt | awk -F "\t" '{$2=($1-1)*4+ 1 ; print $2}' > raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz.1_col.filter.order.txt.order_1 && echo -e "filter_1_col.order.04_get_crick.sh.raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz.1_col.filter.order.txt.order_1\tWed May  1 09:32:09 2024\tless raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz.1_col.filter.order.txt | awk -F \x22\t\x22 \x27{$2=($1-1)*4+ 1 ; print $2}\x27 > raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz.1_col.filter.order.txt.order_1 #ok\n\n\n" >> readme.status.txt
cd .../fastq
python3 /mnt/server1/data2/tangym/p300s/tangym/Analysis_data/auto_spatial_methylation/python_script/fq_line_pick_and_gz.py raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz.1_col.filter.order.txt.order_1 .../fastq/sample_name.raw_R2.fq.gz -o raw_R2.sample_name.extract_R2.crick.extract_all.fq.gz && echo -e "extract_order_all_fq_gz.04_get_crick.sh\tWed May  1 09:32:09 2024\tpython3 /mnt/server1/data2/tangym/p300s/tangym/Analysis_data/auto_spatial_methylation/python_script/fq_line_pick_and_gz.py raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz.1_col.filter.order.txt.order_1 .../fastq/sample_name.raw_R2.fq.gz -o raw_R2.sample_name.extract_R2.crick.extract_all.fq.gz #ok\n\n\n" >> readme.status.txt
cd .../fastq
zcat raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz | awk 'NR%4==1{print $0}' | cut -f1 -d " " | cut -f1 -d "_" > raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz.filter.1_col.txt && echo -e "zcatawkcut.04_get_crick.sh.raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz\tWed May  1 09:32:09 2024\tzcat raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz | awk \x27NR%4==1{print $0}\x27 | cut -f1 -d \x22 \x22 | cut -f1 -d \x22_\x22 > raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz.filter.1_col.txt #ok\n\n\n" >> readme.status.txt
cd .../fastq
zcat .../fastq/sample_name.raw_R1.fq.gz | awk 'NR%4==1{print $0}' | cut -f1 -d " " | cut -f1 -d "_" > sample_name.raw_R1.fq.gz.filter.1_col.txt && echo -e "zcatawkcut.04_get_crick.sh..sample_name.raw_R1.fq.gz\tWed May  1 09:32:09 2024\tzcat .../fastq/sample_name.raw_R1.fq.gz | awk \x27NR%4==1{print $0}\x27 | cut -f1 -d \x22 \x22 | cut -f1 -d \x22_\x22 > sample_name.raw_R1.fq.gz.filter.1_col.txt #ok\n\n\n" >> readme.status.txt
cd .../fastq
less sample_name.raw_R1.fq.gz.filter.1_col.txt | awk 'NR==FNR{a[$1]=NR}NR!=FNR{print a[$1]}' - raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz.filter.1_col.txt | grep -v '^$' > raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz.1_col.filter.order.txt && echo -e "filter_1_col.order.04_get_crick.sh\tWed May  1 09:32:09 2024\tless sample_name.raw_R1.fq.gz.filter.1_col.txt | awk \x27NR==FNR{a[$1]=NR}NR!=FNR{print a[$1]}\x27 - raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz.filter.1_col.txt | grep -v \x27^$\x27 > raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz.1_col.filter.order.txt #ok\n\n\n" >> readme.status.txt
cd .../fastq
less raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz.1_col.filter.order.txt | awk -F "\t" '{$2=($1-1)*4+ 1 ; print $2}' > raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz.1_col.filter.order.txt.order_1 && echo -e "filter_1_col.order.04_get_crick.sh.raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz.1_col.filter.order.txt.order_1\tWed May  1 09:32:09 2024\tless raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz.1_col.filter.order.txt | awk -F \x22\t\x22 \x27{$2=($1-1)*4+ 1 ; print $2}\x27 > raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz.1_col.filter.order.txt.order_1 #ok\n\n\n" >> readme.status.txt
cd .../fastq
python3 /mnt/server1/data2/tangym/p300s/tangym/Analysis_data/auto_spatial_methylation/python_script/fq_line_pick_and_gz.py raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz.1_col.filter.order.txt.order_1 .../fastq/sample_name.raw_R1.fq.gz -o raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all.fq.gz && echo -e "extract_order_all_fq_gz.04_get_crick.sh\tWed May  1 09:32:09 2024\tpython3 /mnt/server1/data2/tangym/p300s/tangym/Analysis_data/auto_spatial_methylation/python_script/fq_line_pick_and_gz.py raw_R2.sample_name.extract_R2.extract_all.crick.fq.gz.1_col.filter.order.txt.order_1 .../fastq/sample_name.raw_R1.fq.gz -o raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all.fq.gz #ok\n\n\n" >> readme.status.txt
#!!!Now is: raw_R2.sample_name.extract_R2.extract_all.fq.gz crick End!!


cd .../bismark/integration
trim_galore --cores 2 -a CTGTCTCTTATACACATCT --clip_R1 34 --three_prime_clip_R1 9 --basename raw_R1.raw_R2.sample_name.extract_R2.extract_watson .../fastq/raw_R1.raw_R2.sample_name.extract_R2.extract_watson.fq.gz && echo -e "bismark.waston.1\tWed May  1 09:32:09 2024\ttrim_galore --cores 8 -a CTGTCTCTTATACACATCT --clip_R1 34 --three_prime_clip_R1 9 --basename raw_R1.raw_R2.sample_name.extract_R2.extract_watson .../fastq/raw_R1.raw_R2.sample_name.extract_R2.extract_watson.fq.gz #ok\n\n\n" >> readme.status.txt
cd .../bismark/integration
/home/tangym/software/Bismark-0.23.0/bismark --unmapped --dovetail --pbat -p 2 /mnt/server1/data2/tangym/xtdisk/tangym/Ref_genomes/index/bismark_index_for_TAB_seq raw_R1.raw_R2.sample_name.extract_R2.extract_watson_trimmed.fq.gz && echo -e "bismark.waston.2\tWed May  1 09:32:09 2024\t/home/tangym/software/Bismark-0.23.0/bismark --unmapped --dovetail --pbat -p 8 /mnt/server1/data2/tangym/xtdisk/tangym/Ref_genomes/index/bismark_index_for_TAB_seq raw_R1.raw_R2.sample_name.extract_R2.extract_watson_trimmed.fq.gz #ok\n\n\n" >> readme.status.txt
cd .../bismark/integration
trim_galore --cores 2 --nextera --three_prime_clip_R1 5 --basename raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_1 raw_R1.raw_R2.sample_name.extract_R2.extract_watson_trimmed.fq.gz_unmapped_reads.fq.gz && echo -e "bismark.waston.3\tWed May  1 09:32:09 2024\ttrim_galore --cores 8 --nextera --three_prime_clip_R1 5 --basename raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_1 raw_R1.raw_R2.sample_name.extract_R2.extract_watson_trimmed.fq.gz_unmapped_reads.fq.gz #ok\n\n\n" >> readme.status.txt
cd .../bismark/integration
mv raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_1_trimmed.fq.gz_unmapped_reads_trimmed.fq.gz raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_1_trimmed.fq.gz && echo -e "bismark.waston.4\tWed May  1 09:32:09 2024\tmv raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_1_trimmed.fq.gz_unmapped_reads_trimmed.fq.gz raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_1_trimmed.fq.gz #ok\n\n\n" >> readme.status.txt
cd .../bismark/integration
/home/tangym/software/Bismark-0.23.0/bismark --unmapped --dovetail -p 2 /mnt/server1/data2/tangym/xtdisk/tangym/Ref_genomes/index/bismark_index_for_TAB_seq raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_1_trimmed.fq.gz && echo -e "bismark.waston.5\tWed May  1 09:32:09 2024\t/home/tangym/software/Bismark-0.23.0/bismark --unmapped --dovetail -p 8 /mnt/server1/data2/tangym/xtdisk/tangym/Ref_genomes/index/bismark_index_for_TAB_seq raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_1_trimmed.fq.gz #ok\n\n\n" >> readme.status.txt
cd .../bismark/integration
trim_galore --cores 2 --nextera --three_prime_clip_R1 5 --basename raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_2 raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_1_trimmed.fq.gz_unmapped_reads.fq.gz && echo -e "bismark.waston.6\tWed May  1 09:32:09 2024\ttrim_galore --cores 8 --nextera --three_prime_clip_R1 5 --basename raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_2 raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_1_trimmed.fq.gz_unmapped_reads.fq.gz #ok\n\n\n" >> readme.status.txt
cd .../bismark/integration
mv raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_2_trimmed.fq.gz_unmapped_reads_trimmed.fq.gz raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_2_trimmed.fq.gz && echo -e "bismark.waston.7\tWed May  1 09:32:09 2024\tmv raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_2_trimmed.fq.gz_unmapped_reads_trimmed.fq.gz raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_2_trimmed.fq.gz #ok\n\n\n" >> readme.status.txt
cd .../bismark/integration
/home/tangym/software/Bismark-0.23.0/bismark --unmapped --dovetail -p 2 /mnt/server1/data2/tangym/xtdisk/tangym/Ref_genomes/index/bismark_index_for_TAB_seq raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_2_trimmed.fq.gz && echo -e "bismark.waston.8\tWed May  1 09:32:09 2024\t/home/tangym/software/Bismark-0.23.0/bismark --unmapped --dovetail -p 8 /mnt/server1/data2/tangym/xtdisk/tangym/Ref_genomes/index/bismark_index_for_TAB_seq raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_2_trimmed.fq.gz #ok\n\n\n" >> readme.status.txt
cd .../bismark/integration
trim_galore --cores 2 --nextera --three_prime_clip_R1 5 --basename raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_3 raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_2_trimmed.fq.gz_unmapped_reads.fq.gz && echo -e "bismark.waston.9\tWed May  1 09:32:09 2024\ttrim_galore --cores 8 --nextera --three_prime_clip_R1 5 --basename raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_3 raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_2_trimmed.fq.gz_unmapped_reads.fq.gz #ok\n\n\n" >> readme.status.txt
cd .../bismark/integration
mv raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_3_trimmed.fq.gz_unmapped_reads_trimmed.fq.gz raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_3_trimmed.fq.gz && echo -e "bismark.waston.10\tWed May  1 09:32:09 2024\tmv raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_3_trimmed.fq.gz_unmapped_reads_trimmed.fq.gz raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_3_trimmed.fq.gz #ok\n\n\n" >> readme.status.txt
cd .../bismark/integration
/home/tangym/software/Bismark-0.23.0/bismark --unmapped --dovetail -p 2 /mnt/server1/data2/tangym/xtdisk/tangym/Ref_genomes/index/bismark_index_for_TAB_seq raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_3_trimmed.fq.gz && echo -e "bismark.waston.11\tWed May  1 09:32:09 2024\t/home/tangym/software/Bismark-0.23.0/bismark --unmapped --dovetail -p 8 /mnt/server1/data2/tangym/xtdisk/tangym/Ref_genomes/index/bismark_index_for_TAB_seq raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_3_trimmed.fq.gz #ok\n\n\n" >> readme.status.txt
cd .../bismark/integration
trim_galore --cores 2 --nextera --three_prime_clip_R1 5 --basename raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_4 raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_3_trimmed.fq.gz_unmapped_reads.fq.gz && echo -e "bismark.waston.12\tWed May  1 09:32:09 2024\ttrim_galore --cores 8 --nextera --three_prime_clip_R1 5 --basename raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_4 raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_3_trimmed.fq.gz_unmapped_reads.fq.gz #ok\n\n\n" >> readme.status.txt
cd .../bismark/integration
mv raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_4_trimmed.fq.gz_unmapped_reads_trimmed.fq.gz raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_4_trimmed.fq.gz && echo -e "bismark.waston.13\tWed May  1 09:32:09 2024\tmv raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_4_trimmed.fq.gz_unmapped_reads_trimmed.fq.gz raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_4_trimmed.fq.gz #ok\n\n\n" >> readme.status.txt
cd .../bismark/integration
/home/tangym/software/Bismark-0.23.0/bismark --unmapped --dovetail -p 2 /mnt/server1/data2/tangym/xtdisk/tangym/Ref_genomes/index/bismark_index_for_TAB_seq raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_4_trimmed.fq.gz && echo -e "bismark.waston.14\tWed May  1 09:32:09 2024\t/home/tangym/software/Bismark-0.23.0/bismark --unmapped --dovetail -p 8 /mnt/server1/data2/tangym/xtdisk/tangym/Ref_genomes/index/bismark_index_for_TAB_seq raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_4_trimmed.fq.gz #ok\n\n\n" >> readme.status.txt
cd .../bismark/integration
trim_galore --cores 2 --nextera --three_prime_clip_R1 5 --basename raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_5 raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_4_trimmed.fq.gz_unmapped_reads.fq.gz && echo -e "bismark.waston.15\tWed May  1 09:32:09 2024\ttrim_galore --cores 8 --nextera --three_prime_clip_R1 5 --basename raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_5 raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_4_trimmed.fq.gz_unmapped_reads.fq.gz #ok\n\n\n" >> readme.status.txt
cd .../bismark/integration
mv raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_5_trimmed.fq.gz_unmapped_reads_trimmed.fq.gz raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_5_trimmed.fq.gz && echo -e "bismark.waston.16\tWed May  1 09:32:09 2024\tmv raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_5_trimmed.fq.gz_unmapped_reads_trimmed.fq.gz raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_5_trimmed.fq.gz #ok\n\n\n" >> readme.status.txt
cd .../bismark/integration
/home/tangym/software/Bismark-0.23.0/bismark --unmapped --dovetail -p 2 /mnt/server1/data2/tangym/xtdisk/tangym/Ref_genomes/index/bismark_index_for_TAB_seq raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_5_trimmed.fq.gz && echo -e "bismark.waston.17\tWed May  1 09:32:09 2024\t/home/tangym/software/Bismark-0.23.0/bismark --unmapped --dovetail -p 8 /mnt/server1/data2/tangym/xtdisk/tangym/Ref_genomes/index/bismark_index_for_TAB_seq raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_5_trimmed.fq.gz #ok\n\n\n" >> readme.status.txt
cd .../bismark/integration
trim_galore --cores 2 -a CTATCTCTTATACACATCT --clip_R1 43 --basename raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all .../fastq/raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all.fq.gz && echo -e "bismark.crick.1\tWed May  1 09:32:09 2024\ttrim_galore --cores 8 -a CTATCTCTTATACACATCT --clip_R1 43 --basename raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all .../fastq/raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all.fq.gz #ok\n\n\n" >> readme.status.txt
cd .../bismark/integration
/home/tangym/software/Bismark-0.23.0/bismark --pbat --unmapped --dovetail --pbat -p 2 /mnt/server1/data2/tangym/xtdisk/tangym/Ref_genomes/index/bismark_index_for_TAB_seq raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_trimmed.fq.gz && echo -e "bismark.crick.2\tWed May  1 09:32:09 2024\t/home/tangym/software/Bismark-0.23.0/bismark --pbat --unmapped --dovetail --pbat -p 8 /mnt/server1/data2/tangym/xtdisk/tangym/Ref_genomes/index/bismark_index_for_TAB_seq raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_trimmed.fq.gz #ok\n\n\n" >> readme.status.txt
cd .../bismark/integration
trim_galore --cores 2 --nextera --three_prime_clip_R1 5 --basename raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_1 raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_trimmed.fq.gz_unmapped_reads.fq.gz && echo -e "bismark.crick.3\tWed May  1 09:32:09 2024\ttrim_galore --cores 8 --nextera --three_prime_clip_R1 5 --basename raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_1 raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_trimmed.fq.gz_unmapped_reads.fq.gz #ok\n\n\n" >> readme.status.txt
cd .../bismark/integration
mv raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_1_trimmed.fq.gz_unmapped_reads_trimmed.fq.gz raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_1_trimmed.fq.gz && echo -e "bismark.crick.4\tWed May  1 09:32:09 2024\tmv raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_1_trimmed.fq.gz_unmapped_reads_trimmed.fq.gz raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_1_trimmed.fq.gz #ok\n\n\n" >> readme.status.txt
cd .../bismark/integration
/home/tangym/software/Bismark-0.23.0/bismark --pbat --unmapped --dovetail -p 2 /mnt/server1/data2/tangym/xtdisk/tangym/Ref_genomes/index/bismark_index_for_TAB_seq raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_1_trimmed.fq.gz && echo -e "bismark.crick.5\tWed May  1 09:32:09 2024\t/home/tangym/software/Bismark-0.23.0/bismark --pbat --unmapped --dovetail -p 8 /mnt/server1/data2/tangym/xtdisk/tangym/Ref_genomes/index/bismark_index_for_TAB_seq raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_1_trimmed.fq.gz #ok\n\n\n" >> readme.status.txt
cd .../bismark/integration
trim_galore --cores 2 --nextera --three_prime_clip_R1 5 --basename raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_2 raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_1_trimmed.fq.gz_unmapped_reads.fq.gz && echo -e "bismark.crick.6\tWed May  1 09:32:09 2024\ttrim_galore --cores 8 --nextera --three_prime_clip_R1 5 --basename raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_2 raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_1_trimmed.fq.gz_unmapped_reads.fq.gz #ok\n\n\n" >> readme.status.txt
cd .../bismark/integration
mv raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_2_trimmed.fq.gz_unmapped_reads_trimmed.fq.gz raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_2_trimmed.fq.gz && echo -e "bismark.crick.7\tWed May  1 09:32:09 2024\tmv raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_2_trimmed.fq.gz_unmapped_reads_trimmed.fq.gz raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_2_trimmed.fq.gz #ok\n\n\n" >> readme.status.txt
cd .../bismark/integration
/home/tangym/software/Bismark-0.23.0/bismark --pbat --unmapped --dovetail -p 2 /mnt/server1/data2/tangym/xtdisk/tangym/Ref_genomes/index/bismark_index_for_TAB_seq raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_2_trimmed.fq.gz && echo -e "bismark.crick.8\tWed May  1 09:32:09 2024\t/home/tangym/software/Bismark-0.23.0/bismark --pbat --unmapped --dovetail -p 8 /mnt/server1/data2/tangym/xtdisk/tangym/Ref_genomes/index/bismark_index_for_TAB_seq raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_2_trimmed.fq.gz #ok\n\n\n" >> readme.status.txt
cd .../bismark/integration
trim_galore --cores 2 --nextera --three_prime_clip_R1 5 --basename raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_3 raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_2_trimmed.fq.gz_unmapped_reads.fq.gz && echo -e "bismark.crick.9\tWed May  1 09:32:09 2024\ttrim_galore --cores 8 --nextera --three_prime_clip_R1 5 --basename raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_3 raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_2_trimmed.fq.gz_unmapped_reads.fq.gz #ok\n\n\n" >> readme.status.txt
cd .../bismark/integration
mv raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_3_trimmed.fq.gz_unmapped_reads_trimmed.fq.gz raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_3_trimmed.fq.gz && echo -e "bismark.crick.10\tWed May  1 09:32:09 2024\tmv raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_3_trimmed.fq.gz_unmapped_reads_trimmed.fq.gz raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_3_trimmed.fq.gz #ok\n\n\n" >> readme.status.txt
cd .../bismark/integration
/home/tangym/software/Bismark-0.23.0/bismark --pbat --unmapped --dovetail -p 2 /mnt/server1/data2/tangym/xtdisk/tangym/Ref_genomes/index/bismark_index_for_TAB_seq raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_3_trimmed.fq.gz && echo -e "bismark.crick.11\tWed May  1 09:32:09 2024\t/home/tangym/software/Bismark-0.23.0/bismark --pbat --unmapped --dovetail -p 8 /mnt/server1/data2/tangym/xtdisk/tangym/Ref_genomes/index/bismark_index_for_TAB_seq raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_3_trimmed.fq.gz #ok\n\n\n" >> readme.status.txt
cd .../bismark/integration
trim_galore --cores 2 --nextera --three_prime_clip_R1 5 --basename raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_4 raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_3_trimmed.fq.gz_unmapped_reads.fq.gz && echo -e "bismark.crick.12\tWed May  1 09:32:09 2024\ttrim_galore --cores 8 --nextera --three_prime_clip_R1 5 --basename raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_4 raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_3_trimmed.fq.gz_unmapped_reads.fq.gz #ok\n\n\n" >> readme.status.txt
cd .../bismark/integration
mv raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_4_trimmed.fq.gz_unmapped_reads_trimmed.fq.gz raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_4_trimmed.fq.gz && echo -e "bismark.crick.13\tWed May  1 09:32:09 2024\tmv raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_4_trimmed.fq.gz_unmapped_reads_trimmed.fq.gz raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_4_trimmed.fq.gz #ok\n\n\n" >> readme.status.txt
cd .../bismark/integration
/home/tangym/software/Bismark-0.23.0/bismark --pbat --unmapped --dovetail -p 2 /mnt/server1/data2/tangym/xtdisk/tangym/Ref_genomes/index/bismark_index_for_TAB_seq raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_4_trimmed.fq.gz && echo -e "bismark.crick.14\tWed May  1 09:32:09 2024\t/home/tangym/software/Bismark-0.23.0/bismark --pbat --unmapped --dovetail -p 8 /mnt/server1/data2/tangym/xtdisk/tangym/Ref_genomes/index/bismark_index_for_TAB_seq raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_4_trimmed.fq.gz #ok\n\n\n" >> readme.status.txt
cd .../bismark/integration
trim_galore --cores 2 --nextera --three_prime_clip_R1 5 --basename raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_5 raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_4_trimmed.fq.gz_unmapped_reads.fq.gz && echo -e "bismark.crick.15\tWed May  1 09:32:09 2024\ttrim_galore --cores 8 --nextera --three_prime_clip_R1 5 --basename raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_5 raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_4_trimmed.fq.gz_unmapped_reads.fq.gz #ok\n\n\n" >> readme.status.txt
cd .../bismark/integration
mv raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_5_trimmed.fq.gz_unmapped_reads_trimmed.fq.gz raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_5_trimmed.fq.gz && echo -e "bismark.crick.16\tWed May  1 09:32:09 2024\tmv raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_5_trimmed.fq.gz_unmapped_reads_trimmed.fq.gz raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_5_trimmed.fq.gz #ok\n\n\n" >> readme.status.txt
cd .../bismark/integration
/home/tangym/software/Bismark-0.23.0/bismark --pbat --unmapped --dovetail -p 2 /mnt/server1/data2/tangym/xtdisk/tangym/Ref_genomes/index/bismark_index_for_TAB_seq raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_5_trimmed.fq.gz && echo -e "bismark.crick.17\tWed May  1 09:32:09 2024\t/home/tangym/software/Bismark-0.23.0/bismark --pbat --unmapped --dovetail -p 8 /mnt/server1/data2/tangym/xtdisk/tangym/Ref_genomes/index/bismark_index_for_TAB_seq raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_5_trimmed.fq.gz #ok\n\n\n" >> readme.status.txt

cd .../bismark/integration
samtools cat -o sample_name.SE.merge.bam raw_R1.raw_R2.sample_name.extract_R2.extract_watson_trimmed_bismark_bt2.bam raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_1_trimmed_bismark_bt2.bam raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_2_trimmed_bismark_bt2.bam raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_3_trimmed_bismark_bt2.bam raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_4_trimmed_bismark_bt2.bam raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_5_trimmed_bismark_bt2.bam raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_trimmed_bismark_bt2.bam raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_1_trimmed_bismark_bt2.bam raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_2_trimmed_bismark_bt2.bam raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_3_trimmed_bismark_bt2.bam raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_4_trimmed_bismark_bt2.bam raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_5_trimmed_bismark_bt2.bam && echo -e "samtools.Cat.bismark.sample_name\tWed May  1 09:32:09 2024\tsamtools cat -o sample_name.SE.merge.bam raw_R1.raw_R2.sample_name.extract_R2.extract_watson_trimmed_bismark_bt2.bam raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_1_trimmed_bismark_bt2.bam raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_2_trimmed_bismark_bt2.bam raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_3_trimmed_bismark_bt2.bam raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_4_trimmed_bismark_bt2.bam raw_R1.raw_R2.sample_name.extract_R2.extract_watson_unmapped_5_trimmed_bismark_bt2.bam raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_trimmed_bismark_bt2.bam raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_1_trimmed_bismark_bt2.bam raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_2_trimmed_bismark_bt2.bam raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_3_trimmed_bismark_bt2.bam raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_4_trimmed_bismark_bt2.bam raw_R1.raw_R2.sample_name.extract_R2.crick.extract_all_unmapped_5_trimmed_bismark_bt2.bam #ok\n\n\n" >> readme.status.txt
cd .../fastq
gunzip -c sample_name.extract_R1.fq.gz > sample_name_spatial_1.extract.txt && echo -e "sample_name_pre_split.sh.1\tWed May  1 09:32:09 2024\tgunzip -c sample_name.extract_R1.fq.gz > sample_name_spatial_1.extract.txt #ok\n\n\n" >> readme.status.txt
cd .../fastq
less sample_name_spatial_1.extract.txt | awk 'NR%4==1{print $0}' > sample_name_spatial_1.extract_index.txt && echo -e "sample_name_pre_split.sh.2\tWed May  1 09:32:09 2024\tless sample_name_spatial_1.extract.txt | awk \x27NR%4==1{print $0}\x27 > sample_name_spatial_1.extract_index.txt #ok\n\n\n" >> readme.status.txt
cd .../bismark/integration
samtools view -H sample_name.SE.merge.bam > header.txt && echo -e "sample_name_pre_split.sh.3\tWed May  1 09:32:09 2024\tsamtools view -H sample_name.SE.merge.bam > header.txt #ok\n\n\n" >> readme.status.txt
cd .../bismark/integration
samtools view sample_name.SE.merge.bam | awk '{print $1,$14}' | sed 's/\.//g' > sample_name.bam.step1.txt && echo -e "sample_name_pre_split.sh.4\tWed May  1 09:32:09 2024\tsamtools view sample_name.SE.merge.bam | awk \x27{print $1,$14}\x27 | sed \x27s/\.//g\x27 > sample_name.bam.step1.txt #ok\n\n\n" >> readme.status.txt
cd .../bismark/integration
wc -l sample_name.bam.step1.txt > Filter.bam.summary && echo -e "sample_name_pre_split.sh.5\tWed May  1 09:32:09 2024\twc -l sample_name.bam.step1.txt > Filter.bam.summary #ok\n\n\n" >> readme.status.txt
cd .../bismark/integration
cat sample_name.bam.step1.txt | awk -F [":"," "] '{print $13}' | sed s/[X,H,Z]/X/g > sample_name.bam.step2.txt && echo -e "sample_name_pre_split.sh.6\tWed May  1 09:32:09 2024\tcat sample_name.bam.step1.txt | awk -F [\x22:\x22,\x22 \x22] \x27{print $13}\x27 | sed s/[X,H,Z]/X/g > sample_name.bam.step2.txt #ok\n\n\n" >> readme.status.txt
cd .../bismark/integration
paste sample_name.bam.step1.txt sample_name.bam.step2.txt > sample_name.bam.step3.txt && echo -e "sample_name_pre_split.sh.7\tWed May  1 09:32:09 2024\tpaste sample_name.bam.step1.txt sample_name.bam.step2.txt > sample_name.bam.step3.txt #ok\n\n\n" >> readme.status.txt
cd .../bismark/integration
cat sample_name.bam.step3.txt | awk -F ["\t"," ",":"] '{print $1":"$2":"$3":"$4":"$5":"$6":"$7":"$8":"$9":"$10" "$11":"$12":"$14}' > sample_name.bam.step4.txt && echo -e "sample_name_pre_split.sh.8\tWed May  1 09:32:09 2024\tcat sample_name.bam.step3.txt | awk -F [\x22\t\x22,\x22 \x22,\x22:\x22] \x27{print $1\x22:\x22$2\x22:\x22$3\x22:\x22$4\x22:\x22$5\x22:\x22$6\x22:\x22$7\x22:\x22$8\x22:\x22$9\x22:\x22$10\x22 \x22$11\x22:\x22$12\x22:\x22$14}\x27 > sample_name.bam.step4.txt #ok\n\n\n" >> readme.status.txt
cd .../bismark/integration
grep -v XXX sample_name.bam.step4.txt | awk -F " " '{print $1}' > sample_name.bam.step5.txt && echo -e "sample_name_pre_split.sh.9\tWed May  1 09:32:09 2024\tgrep -v XXX sample_name.bam.step4.txt | awk -F \x22 \x22 \x27{print $1}\x27 > sample_name.bam.step5.txt #ok\n\n\n" >> readme.status.txt
cd .../bismark/integration
wc -l sample_name.bam.step5.txt >> Filter.bam.summary && echo -e "sample_name_pre_split.sh.10\tWed May  1 09:32:09 2024\twc -l sample_name.bam.step5.txt >> Filter.bam.summary #ok\n\n\n" >> readme.status.txt
cd .../bismark/integration
java -jar /home/tangym/software/picard.jar FilterSamReads I=sample_name.SE.merge.bam O=sample_name.SE.merge.filtered.bam READ_LIST_FILE=sample_name.bam.step5.txt FILTER=includeReadList VALIDATION_STRINGENCY=LENIENT && echo -e "sample_name_pre_split.sh.11\tWed May  1 09:32:09 2024\tjava -jar /home/tangym/software/picard.jar FilterSamReads I=sample_name.SE.merge.bam O=sample_name.SE.merge.filtered.bam READ_LIST_FILE=sample_name.bam.step5.txt FILTER=includeReadList VALIDATION_STRINGENCY=LENIENT #ok\n\n\n" >> readme.status.txt


cd .../bismark/split
python3 /mnt/server1/data2/tangym/p300s/tangym/Analysis_data/auto_spatial_methylation/python_script/bam_split_to_cell.py .../bismark/integration/sample_name.SE.merge.filtered.bam .../fastq/sample_name.extract_R2.fq.gz -e each_part -ref /mnt/server1/data2/tangym/p300s/tangym/Analysis_data/auto_spatial_methylation/bins/dir_ref_barcode_one_row.txt  && echo -e "bam_split_to_cell.py.split\tWed May  1 09:32:09 2024\tpython3 /mnt/server1/data2/tangym/p300s/tangym/Analysis_data/auto_spatial_methylation/python_script/bam_split_to_cell.py .../bismark/integration/sample_name.SE.merge.filtered.bam .../fastq/sample_name.extract_R2.fq.gz -e 500 #ok\n\n\n" >> readme.status.txt


let "value=all_cells/each_part+1"

for i in `seq ${value} | paste -s `

do

mkdir -p .../bismark/split/part${i}
echo !!!Waring: This step must after split by bam_split_to_cell.py run...
cd .../bismark/split/part${i}
python3 /mnt/server1/data2/tangym/p300s/tangym/Analysis_data/auto_spatial_methylation/python_script/bismark_cell_calculate.py . && echo -e "bismark_cell_calculate.py.part\${i}\tWed May  1 09:32:09 2024\tpython3 /mnt/server1/data2/tangym/p300s/tangym/Analysis_data/auto_spatial_methylation/python_script/bismark_cell_calculate.py . #ok\n\n\n" >> readme.status.txt


mkdir -p .../bismark/split/part${i}
cd .../bismark/split/part${i}
python3 /mnt/server1/data2/tangym/p300s/tangym/Analysis_data/auto_spatial_methylation/python_script/bismark_cell_calculate.py . -cm && echo -e "bismark_cell_calculate.py.cm..part\${i}\tWed May  1 09:32:09 2024\tpython3 /mnt/server1/data2/tangym/p300s/tangym/Analysis_data/auto_spatial_methylation/python_script/bismark_cell_calculate.py . -cm #ok\n\n\n" >> readme.status.txt


done

wait



cd .../bismark/split
cat ./part1/methylation_matrix.txt ./part2/methylation_matrix.txt ./part3/methylation_matrix.txt ./part4/methylation_matrix.txt ./part5/methylation_matrix.txt ./part6/methylation_matrix.txt ./part7/methylation_matrix.txt ./part8/methylation_matrix.txt ./part9/methylation_matrix.txt ./part10/methylation_matrix.txt ./part11/methylation_matrix.txt ./part12/methylation_matrix.txt ./part13/methylation_matrix.txt ./part14/methylation_matrix.txt ./part15/methylation_matrix.txt ./part16/methylation_matrix.txt ./part17/methylation_matrix.txt ./part18/methylation_matrix.txt ./part19/methylation_matrix.txt > merge.methylation_matrix.txt && echo -e "bismark_cell_calculate.py.cm.methylation_matrix.txt\tWed May  1 09:32:09 2024\tcat ./part1/methylation_matrix.txt ./part2/methylation_matrix.txt ./part3/methylation_matrix.txt ./part4/methylation_matrix.txt ./part5/methylation_matrix.txt ./part6/methylation_matrix.txt ./part7/methylation_matrix.txt ./part8/methylation_matrix.txt ./part9/methylation_matrix.txt ./part10/methylation_matrix.txt ./part11/methylation_matrix.txt ./part12/methylation_matrix.txt ./part13/methylation_matrix.txt ./part14/methylation_matrix.txt ./part15/methylation_matrix.txt ./part16/methylation_matrix.txt ./part17/methylation_matrix.txt ./part18/methylation_matrix.txt ./part19/methylation_matrix.txt > merge.methylation_matrix.txt #ok\n\n\n" >> readme.status.txt
cd .../bismark/split
cat ./part1/coverage_matrix.txt ./part2/coverage_matrix.txt ./part3/coverage_matrix.txt ./part4/coverage_matrix.txt ./part5/coverage_matrix.txt ./part6/coverage_matrix.txt ./part7/coverage_matrix.txt ./part8/coverage_matrix.txt ./part9/coverage_matrix.txt ./part10/coverage_matrix.txt ./part11/coverage_matrix.txt ./part12/coverage_matrix.txt ./part13/coverage_matrix.txt ./part14/coverage_matrix.txt ./part15/coverage_matrix.txt ./part16/coverage_matrix.txt ./part17/coverage_matrix.txt ./part18/coverage_matrix.txt ./part19/coverage_matrix.txt > merge.coverage_matrix.txt && echo -e "bismark_cell_calculate.py.cm.coverage_matrix.txt\tWed May  1 09:32:09 2024\tcat ./part1/coverage_matrix.txt ./part2/coverage_matrix.txt ./part3/coverage_matrix.txt ./part4/coverage_matrix.txt ./part5/coverage_matrix.txt ./part6/coverage_matrix.txt ./part7/coverage_matrix.txt ./part8/coverage_matrix.txt ./part9/coverage_matrix.txt ./part10/coverage_matrix.txt ./part11/coverage_matrix.txt ./part12/coverage_matrix.txt ./part13/coverage_matrix.txt ./part14/coverage_matrix.txt ./part15/coverage_matrix.txt ./part16/coverage_matrix.txt ./part17/coverage_matrix.txt ./part18/coverage_matrix.txt ./part19/coverage_matrix.txt > merge.coverage_matrix.txt #ok\n\n\n" >> readme.status.txt
cd .../bismark/split
cat ./part1/depth_matrix.txt ./part2/depth_matrix.txt ./part3/depth_matrix.txt ./part4/depth_matrix.txt ./part5/depth_matrix.txt ./part6/depth_matrix.txt ./part7/depth_matrix.txt ./part8/depth_matrix.txt ./part9/depth_matrix.txt ./part10/depth_matrix.txt ./part11/depth_matrix.txt ./part12/depth_matrix.txt ./part13/depth_matrix.txt ./part14/depth_matrix.txt ./part15/depth_matrix.txt ./part16/depth_matrix.txt ./part17/depth_matrix.txt ./part18/depth_matrix.txt ./part19/depth_matrix.txt > merge.depth_matrix.txt && echo -e "bismark_cell_calculate.py.cm.depth_matrix.txt\tWed May  1 09:32:09 2024\tcat ./part1/depth_matrix.txt ./part2/depth_matrix.txt ./part3/depth_matrix.txt ./part4/depth_matrix.txt ./part5/depth_matrix.txt ./part6/depth_matrix.txt ./part7/depth_matrix.txt ./part8/depth_matrix.txt ./part9/depth_matrix.txt ./part10/depth_matrix.txt ./part11/depth_matrix.txt ./part12/depth_matrix.txt ./part13/depth_matrix.txt ./part14/depth_matrix.txt ./part15/depth_matrix.txt ./part16/depth_matrix.txt ./part17/depth_matrix.txt ./part18/depth_matrix.txt ./part19/depth_matrix.txt > merge.depth_matrix.txt #ok\n\n\n" >> readme.status.txt

cd .../plot
/home/liulab/anaconda3/envs/tangym_workspace/bin/Rscript /mnt/server1/data2/tangym/p300s/tangym/Analysis_data/auto_spatial_methylation/bins/4_plot_methylation_matrix.R -p .../plot -n .../bismark/split/merge.methylation_matrix.txt -b /mnt/server1/data2/tangym/p300s/tangym/Analysis_data/auto_spatial_methylation/bins/combine_barcode.round2round1_index1_index2.v3_big_methylation.txt && echo -e "4_plot_methylation_matrix.R.plot\tWed May  1 09:32:09 2024\t/home/liulab/anaconda3/envs/tangym_workspace/bin/Rscript /mnt/server1/data2/tangym/p300s/tangym/Analysis_data/auto_spatial_methylation/bins/4_plot_methylation_matrix.R -p .../plot -n .../bismark/split/merge.methylation_matrix.txt -b /mnt/server1/data2/tangym/p300s/tangym/Analysis_data/spatial_methylation/combine_barcode.round2round1_index1_index2.v3_big_methylation.txt #ok\n\n\n" >> readme.status.txt
cd .../plot
/home/liulab/anaconda3/envs/tangym_workspace/bin/Rscript /mnt/server1/data2/tangym/p300s/tangym/Analysis_data/auto_spatial_methylation/bins/5_plot_coverage_matrix.R -p .../plot -n .../bismark/split/merge.coverage_matrix.txt -b /mnt/server1/data2/tangym/p300s/tangym/Analysis_data/auto_spatial_methylation/bins/combine_barcode.round2round1_index1_index2.v3_big_methylation.txt && echo -e "5_plot_coverage_matrix.R.plot\tWed May  1 09:32:09 2024\t/home/liulab/anaconda3/envs/tangym_workspace/bin/Rscript /mnt/server1/data2/tangym/p300s/tangym/Analysis_data/auto_spatial_methylation/bins/5_plot_coverage_matrix.R -p .../plot -n .../bismark/split/merge.coverage_matrix.txt -b /mnt/server1/data2/tangym/p300s/tangym/Analysis_data/spatial_methylation/combine_barcode.round2round1_index1_index2.v3_big_methylation.txt #ok\n\n\n" >> readme.status.txt
cd .../plot
/home/liulab/anaconda3/envs/tangym_workspace/bin/Rscript /mnt/server1/data2/tangym/p300s/tangym/Analysis_data/auto_spatial_methylation/bins/6_plot_depth_matrix.R -p .../plot -n .../bismark/split/merge.depth_matrix.txt -b /mnt/server1/data2/tangym/p300s/tangym/Analysis_data/auto_spatial_methylation/bins/combine_barcode.round2round1_index1_index2.v3_big_methylation.txt && echo -e "6_plot_depth_matrix.R.plot\tWed May  1 09:32:09 2024\t/home/liulab/anaconda3/envs/tangym_workspace/bin/Rscript /mnt/server1/data2/tangym/p300s/tangym/Analysis_data/auto_spatial_methylation/bins/6_plot_depth_matrix.R -p .../plot -n .../bismark/split/merge.depth_matrix.txt -b /mnt/server1/data2/tangym/p300s/tangym/Analysis_data/spatial_methylation/combine_barcode.round2round1_index1_index2.v3_big_methylation.txt #ok\n\n\n" >> readme.status.txt
